File: nuget/nugetpackage/Check.cs
using System.Collections;
using System.Text.RegularExpressions;

namespace nugetpackage;

public class Check
{
    public static class Email
    {
        static ArrayList validDomainNames = new ArrayList();
        static ArrayList validDomainExtensions = new ArrayList();
        static ArrayList validDomains = new ArrayList();
        static bool shouldUseFullDomain = false;
        public static void AddValidDomainName(string str)
        {
            validDomainNames.Add(str);
        }
        public static void AddValidDomainExtension(string str)
        {
            validDomainExtensions.Add(str);
        }
        public static void AddValidDomain(string str)
        {
            validDomains.Add(str);
        }
        public static void ShouldUseFullDomain()
        {
            shouldUseFullDomain = true;
        }
        public static void ShouldUseFullDomain(bool boolean)
        {
            shouldUseFullDomain = boolean;
        }
        public static bool IsValid(string str)
        {
            if (shouldUseFullDomain)
            {
                try
                {
                    string[] domain = str.Split('@');
                    return validDomains.Contains(domain[1]);
                }
                catch
                {
                    return false;
                }
            } else
            {
                try
                {
                    string[] domain = str.Split('@');
                    string domainName = domain[1].Split('.')[0];
                    string domainExtension = domain[1].Split('.')[1];
                    return validDomainNames.Contains(domainName) && validDomainExtensions.Contains(domainExtension);
                }
                catch
                {
                    return false;
                }
            }
        }
    }
    public static bool IsAValidPhilippineMobileNumber(string str)
    {
        string pattern = @"^(?:09|\+639|639)\d{9}$";
        return Regex.IsMatch(Regex.Replace(str, @"[\s\-\(\)]", ""), pattern);
    }
    public static bool IsAllNumbers(string str)
    {
        bool value = true;
        string numbers = "0123456789";
        foreach (char a in str)
        {
            if (!numbers.Contains(a))
            {
                value = false;
            }
        }
        return value;
    }
    public static bool HasNumbers(string str)
    {
        string numbers = "0123456789";
        foreach (char a in str)
        {
            foreach (char b in numbers)
            {
                if (a == b)
                {
                    return true;
                }
            }
        }
        return false;
    }
    public static bool HasSymbols(string str)
    {
        string symbols = "~`!@#$%^&*()_+-=[]{}\\|'\";:,.<>/?";
        foreach (char a in str)
        {
            foreach (char b in symbols)
            {
                if (a == b)
                {
                    return true;
                }
            }
        }
        return false;
    }
    public static bool HasSpaces(string str)
    {
        foreach (char a in str)
        {
            if (a == ' ')
            {
                return true;
            }
        }
        return false;
    }
}

File: pypi/pythonbundle/check.py
import re
import base64
import os
import pkgutil

class Check:
    class Email:
        valid_domain_names = []
        valid_domain_extensions = []
        valid_domains =[]
        _should_use_full_domain = False

        @classmethod
        def add_valid_domain_name(cls, string: str):
            cls.valid_domain_names.append(string)

        @classmethod
        def add_valid_domain_extension(cls, string: str):
            cls.valid_domain_extensions.append(string)

        @classmethod
        def add_valid_domain(cls, string: str):
            cls.valid_domains.append(string)

        @classmethod
        def should_use_full_domain(cls, boolean: bool = True):
            cls._should_use_full_domain = boolean

        @classmethod
        def is_valid(cls, string: str) -> bool:
            if cls._should_use_full_domain:
                try:
                    domain = string.split('@')
                    return domain[1] in cls.valid_domains
                except Exception:
                    return False
            else:
                try:
                    domain = string.split('@')
                    domain_parts = domain[1].split('.')
                    domain_name = domain_parts[0]
                    domain_extension = domain_parts[1]
                    return (domain_name in cls.valid_domain_names and 
                            domain_extension in cls.valid_domain_extensions)
                except Exception:
                    return False

    @staticmethod
    def is_a_valid_philippine_mobile_number(string: str) -> bool:
        pattern = r"^(?:09|\+639|639)\d{9}$"
        cleaned = re.sub(r"[\s\-\(\)]", "", string)
        return bool(re.match(pattern, cleaned))

    @staticmethod
    def is_all_numbers(string: str) -> bool:
        numbers = "0123456789"
        return all(char in numbers for char in string)

    @staticmethod
    def has_numbers(string: str) -> bool:
        numbers = "0123456789"
        return any(char in numbers for char in string)

    @staticmethod
    def has_symbols(string: str) -> bool:
        symbols = "~`!@#$%^&*()_+-=[]{}\\|'\";:,.<>/?"
        return any(char in symbols for char in string)

    @staticmethod
    def has_spaces(string: str) -> bool:
        return ' ' in string
